File: VRSampleApp/VRSampleApp/Classes/Utils/PlayerSelectionOption.m
//
//  PlayerSelectionOption.m
//  VRSampleApp
//
//  Copyright © 2017 Ooyala Inc. All rights reserved.
//

#import "PlayerSelectionOption.h"


@implementation PlayerSelectionOption

  - (id)initWithTitle:(NSString *)title embedCode:(NSString *)embedCode pcode:(NSString *)pcode  domain:(NSString *)domain {
    if (self = [super init]) {
      self.title = title;
      self.embedCode = embedCode;
      self.pcode = pcode;
      self.domain = domain;
    }
    return self;
  }

- (id)initWithTitle:(NSString *)title embedCode:(NSString *)embedCode pcode:(NSString *)pcode {
  if (self = [super init]) {
    self.title = title;
    self.embedCode = embedCode;
    self.pcode = pcode;
    self.domain = @"http://www.ooyala.com";
  }
  return self;
}

- (id)initWithTitle:(NSString *)title embedCode:(NSString *)embedCode {
  if (self = [super init]) {
    self.title = title;
    self.embedCode = embedCode;
    self.pcode = @"BzY2syOq6kIK6PTXN7mmrGVSJEFj";
    self.domain = @"http://www.ooyala.com";
  }
  return self;
}
  
@end

File: PlaybackLab/ControlsLockScreenSampleApp/ControlsLockScreenSampleApp/Models/PlayerSelectionOption.swift
//
//  PlayerSelectionOption.swift
//  ControlsLockScreenSampleApp
//
//  Copyright © 2017 Ooyala. All rights reserved.
//

import Foundation

class PlayerSelectionOption {
  
  public private(set) var domain: OOPlayerDomain
  public private(set) var pcode: String
  public private(set) var embedCode: String
  public private(set) var title: String
  public private(set) var thumbnailURL: URL
  public private(set) var embedTokenGenerator: OOEmbedTokenGenerator?

  init(pcode: String, embedCode: String, title: String, thumbnailURL: URL,
       domain: OOPlayerDomain, embedTokenGenerator: OOEmbedTokenGenerator? = nil) {
    self.domain = domain
    self.pcode = pcode
    self.embedCode = embedCode
    self.title = title
    self.thumbnailURL = thumbnailURL
    self.embedTokenGenerator = embedTokenGenerator
  }
  
}
